src/app/hooks/useClickTracker/index.test.jsx

```diff
@@ -0,0 +1,320 @@
+/* eslint-disable react/prop-types */
+import React from 'react';
+import { renderHook } from '@testing-library/react-hooks';
+import { render, act } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import useClickTracker from '.';
+import pidginData from './fixtureData/tori-51745682.json';
+import zhongwenData from './fixtureData/chinese-news-49631219-trad.json';
+import { RequestContextProvider } from '#contexts/RequestContext';
+import { ServiceContextProvider } from '#contexts/ServiceContext';
+import { STORY_PAGE } from '#app/routes/utils/pageTypes';
+
+beforeEach(() => {
+  jest.clearAllMocks();
+});
+
+const urlToObject = url => {
+  const { origin, pathname, searchParams } = new URL(url);
+
+  return {
+    origin,
+    pathname,
+    searchParams: Object.fromEntries(searchParams),
+  };
+};
+
+delete window.location;
+window.location = { href: 'http://bbc.com/pidgin/tori-51745682' };
+process.env.SIMORGH_ATI_BASE_URL = 'https://logws1363.ati-host.net?';
+
+const defaultProps = {
+  pageData: pidginData,
+  componentName: 'brand',
+};
+
+const WithContexts = ({
+  children,
+  variant,
+  service = 'pidgin',
+  pathname = '/pidgin/tori-51745682',
+}) => (
+  <RequestContextProvider
+    bbcOrigin="https://www.test.bbc.com"
+    pageType={STORY_PAGE}
+    isAmp={false}
+    pathname={pathname}
+    service={service}
+    variant={variant}
+  >
+    <ServiceContextProvider service={service}>
+      {children}
+    </ServiceContextProvider>
+  </RequestContextProvider>
+);
+
+const TestComponentContainerContainer = () => {
+  const clickRef = useClickTracker({
+    pageData: pidginData,
+    componentName: 'header',
+  });
+
+  return (
+    <div ref={clickRef}>
+      <TestComponentContainer hookProps={defaultProps} />
+    </div>
+  );
+};
+
+const TestComponentContainer = ({ hookProps }) => {
+  const clickRef = useClickTracker(hookProps);
+  return (
+    <div>
+      <TestComponent ref={clickRef} />
+    </div>
+  );
+};
+
+const TestComponent = React.forwardRef((_, ref) => {
+  return (
+    <div id="test-component" ref={ref}>
+      Containing Div
+      <a href="https://www.test.bbc.com">Testing Link</a>
+      <button type="button" id="test-button">
+        Button
+      </button>
+    </div>
+  );
+});
+
+// eslint-disable-next-line react/prop-types
+const wrapper = ({ children }) => <WithContexts>{children}</WithContexts>;
+
+describe('Click tracking', () => {
+  it('should return a ref', () => {
+    const { result } = renderHook(() => useClickTracker(defaultProps), {
+      wrapper,
+    });
+
+    const { getByText } = render(
+      <WithContexts>
+        <TestComponent ref={result.current} />
+      </WithContexts>,
+    );
+
+    expect(result.current.current).toBe(getByText('Containing Div'));
```

Comment:
sorry to be a pain but this selector might be better off using https://testing-library.com/docs/queries/bytestid rather than having to add text to the DOM to query

Reply:
Yeah good idea, done 👍